### PR TITLE
improve support to coexist with standard scheduler 

### PR DESCRIPTION
**Pull request by ffromani**
Add API field to configure the resync strategy.

We used to configure the cache resync code to:
1. react to any foreign pod (= pod not scheduled by this scheduler)
2. consider all the pods when computing the expected node state

But thinking about it, only pods whose containers require exclusive resources are relevant to the node state, because only exclusive resources can have NUMA affinity.

We retroactively dub this configuration "aggressive" because it causes recomputation when any change is detected, even unnecessary ones.

Recomputing the state often creates more work on the scheduler process, but most notably slows down the scheduling because we need to wait more for resyncs, being the current state invalidated (because of aggressive detection).

To improve in both areas (resource consumption but most notably scheduling latency) we add a new mode, called "relaxed" on which we leverage the scheduler options to react and consider only pods whose containers require exclusive resources.

The new "relaxed" mode is enabled by default, and the old behavior can be re-enabled with a configuration option.

### CODE REVIEW

**Comment by ffromani** [comment]
missing e2e tests
missing docs/commit messages

**Comment by ffromani** [comment]
the podresources socket failure is very suspicious, one last retest to rule out CI flakes (the fact the bot takes minutes to react warrants the test)

**Comment by ffromani** [comment]
the PR is now ready for review

File: test/utils/noderesourcetopologies/noderesourcetopologies.go [hunk 1]
```diff
@@ -508,3 +508,45 @@ func ResourceInfoListToResourceList(ri nrtv1alpha2.ResourceInfoList) corev1.Reso
...
+}
+
+func ResourceInfoProvidingAtMost(resources []nrtv1alpha2.ResourceInfo, resName string, resQty resource.Quantity) bool {
+	zeroQty := resource.MustParse("0")
+	zoneQty, ok := FindResourceAvailableByName(resources, resName)
+	klog.Infof("  +--> checking if resources include %q in (0, %s] (zoneQty=%s found=%v)", resName, resQty.String(), zoneQty.String(), ok)
+	if !ok {
+		return false
+	}
+	if zoneQty.Cmp(zeroQty) <= 0 {
```
Comment (Tal-or) [inline test/utils/noderesourcetopologies/noderesourcetopologies.go:545]: You can use `zoneQty.IsZero`

File: test/e2e/serial/tests/scheduler_cache.go [hunk 2]
```diff
@@ -612,12 +597,25 @@ var _ = Describe("[serial][scheduler][cache][tier1] scheduler cache", Label("sch
 				// NRT updater, resync loop, scheduler retry loop.
 				failedPods, updatedPods = wait.With(fxt.Client).Timeout(5*time.Minute).ForPodListAllRunning(context.TODO(), expectedRunningPods)
 				Expect(len(updatedPods)).To(Equal(hostsRequired))
+				dumpFailedPodInfo(fxt, failedPods)
```
Comment (Tal-or) [inline test/e2e/serial/tests/scheduler_cache.go:600]: In case of failed assertion in the line above, we won't get this information.

**Comment by Tal-or** [comment]
Thanks for the PR, small comments from my side, but the general direction LGTM

File: test/utils/noderesourcetopologies/noderesourcetopologies.go (hunk 1 above)
Comment (ffromani) [reply test/utils/noderesourcetopologies/noderesourcetopologies.go:545]: thanks, fixed

File: test/e2e/serial/tests/scheduler_cache.go (hunk 2 above)
Comment (ffromani) [reply test/e2e/serial/tests/scheduler_cache.go:600]: good catch, let me address this

**Comment by ffromani** [comment]
> don't we need to backport this commit: [e658290](https://github.com/openshift-kni/numaresources-operator/commit/e6582903af1d5552d9d56961d481eee5d6d5aae6)?

That commit changes the default behavior, so it should backported very carefully. We don't for now, because this feature is meant for newer NUMA-scheduler stacks

**Pull request merged**